pkg/proxy/ipvs/proxier.go

```diff
@@ -1009,30 +1033,48 @@ func (proxier *Proxier) syncProxyRules() {
 
 			// Nodeports need SNAT, unless they're local.
 			// ipset call
-			if !svcInfo.OnlyNodeLocalEndpoints {
-				entry = &utilipset.Entry{
-					// No need to provide ip info
-					Port:     svcInfo.NodePort,
-					Protocol: protocol,
-					SetType:  utilipset.BitmapPort,
+			entry = &utilipset.Entry{
+				// No need to provide ip info
+				Port:     svcInfo.NodePort,
+				Protocol: protocol,
+				SetType:  utilipset.BitmapPort,
+			}
+			var nodePortSet *IPSet
```

Comment:
I'm confused on what is happening here. Why do you initialize a new IPSet and then set that to either the existing TCP or UDP nodeport set?

Reply:
This part is to chose the ipset(`KUBE-NODE-PORT-TCP/KUBE-NODE-PORT-UDP`) to insert the entry. Haven't initialized a new ipset, just declared a ipset pointer.